doc/Developer-Guide.md

```diff
@@ -1,38 +1,92 @@
+
+Since actions are named beans, they can easily be added to the YAML configuration file to be executable in the different modules. This is implemented in the [ActionService](../kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/services/ActionService.java). Requested actions to be performed at a later time are stored in a simple database (see the database model below) and performed using the CLI.
+
+Requested actions are stored in a simple database where the action always is connected to a work and can have any number of parameters (see the database model below).
 
 ### Implementing actions
 
-To implement a new action, you implement the [IAction](https://github.com/tuub/kitodo-mediaserver/blob/master/kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/api/IAction.java) interface.
+To implement a new action, you implement the [IAction](../kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/api/IAction.java) interface. To make this action visible to Spring, you simply annotate it as a component and give it a unique name:
+
+```java
+@Component("myUniqueAction")
+public class MyAction implements IAction {
+
+    @Override
+    public Object perform(Work work, Map<String, String> parameter) throws Exception {
+        // do something
+        return null;
+    }
+}
+```
+
+The action will then automatically be instantiated as a Spring bean and can be added in thconfiguration yaml using the name `myUniqueAction`.
+
+It is encouraged, to implement actions in the core module and make a pull request, so that they can be used by others as well. However, if you know, your action can only be used locally, you may just as well put it in the local module.
 
-To be able to perform the action, you define a Spring Bean for it with a speaking name. Using this name, the action can then i.e. be configured to be performed on import. TODO
 
 ### Implementing conversion classes
 
-To implement a class for file conversion, you have to implement the [IConverter](https://github.com/tuub/kitodo-mediaserver/blob/master/kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/api/IConverter.java) interface. You will find implementation examples [here](https://github.com/tuub/kitodo-mediaserver/tree/master/kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/conversion). 
+To implement a class for file conversion, you have to implement the [IConverter](../kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/api/IConverter.java) interface. You will find implementation examples [here](../kitodo-mediaserver/tree/master/kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/conversion).
```

Comment:
The examples path should be `../kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/conversion`